app/src/scripts/dataset/dataset.file-display.jsx

```diff
@@ -22,11 +22,42 @@ class FileDisplay extends Reflux.Component {
   constructor() {
     super()
     refluxConnect(this, datasetStore, 'datasets')
+    this.state = {
+      fileRequested: false,
+    }
   }
 
-  render() {
+  componentWillReceiveProps() {
```

Comment:
This could just be a route.

Reply:
/datasetUrl/file-display/filename instead of /datasetUrl/file-display?file=filename?

Reply:
Yes, since each file is its own page in this case.